static/app/views/performance/newTraceDetails/traceModels/traceTree.tsx

```diff
@@ -1387,8 +1484,20 @@ export class TraceTree extends TraceTreeEventDispatcher {
       this.list.splice(index + 1, TraceTree.VisibleChildren(node).length);
 
       node.expanded = expanded;
+
+      // When eap-transaction nodes are collapsed, they still render transactions as visible children.
+      // Reparent the transactions from under the eap-spans in the expanded state, to under the closest eap-transaction
+      // in the collapsed state.
+      if (isEAPTransactionNode(node)) {
+        if (isEAPTransactionNode(node)) {
```

Comment:
seems like a redundant line?

Reply:
Yep, must've made it in during the conflict resolving